src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -60,6 +60,69 @@ public function add($item)
     }
 
     /**
+     * Get an array with the values of a given key.
+     *
+     * @param  string  $value
+     * @param  string|null  $key
+     * @return \Illuminate\Support\Collection
+     */
+    public function pluck($value, $key = null)
+    {
+        return new BaseCollection(array_pluck($this->items, $value, $key));
+    }
+
+    /**
+     * Get the keys of the collection items.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function keys()
+    {
+        return new BaseCollection(array_keys($this->items));
+    }
+
+    /**
+     * Zip the collection together with one or more arrays.
+     *
+     * @param  mixed ...$items
+     * @return \Illuminate\Support\Collection
+     */
+    public function zip($items)
+    {
+        return new BaseCollection(call_user_func_array('parent::zip', func_get_args()));
+    }
+
+    /**
+     * Collapse the collection of items into a single array.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function collapse()
+    {
+        return new BaseCollection(array_collapse($this->items));
+    }
+
+    /**
+     * Get a flattened array of the items in the collection.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function flatten()
+    {
+        return new BaseCollection(array_flatten($this->items));
+    }
+
+    /**
+     * Flip the items in the collection.
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    public function flip()
+    {
+        return new BaseCollection(array_flip($this->items));
```

Comment:
I think you can use the `make()` method like `return BaseCollection::make($this->items)->flip()` to save yourself from writing implementation here?